src/Operators/integrals.jl

```diff
@@ -155,6 +170,94 @@ function _column_integral_indefinite!(
     end
 end
 
+dual_space(space::Spaces.FaceExtrudedFiniteDifferenceSpace) =
+    Spaces.CenterExtrudedFiniteDifferenceSpace(space)
+dual_space(space::Spaces.CenterExtrudedFiniteDifferenceSpace) =
+    Spaces.FaceExtrudedFiniteDifferenceSpace(space)
+
+dual_space(space::Spaces.FaceFiniteDifferenceSpace) =
+    Spaces.CenterFiniteDifferenceSpace(space)
+dual_space(space::Spaces.CenterFiniteDifferenceSpace) =
+    Spaces.FaceFiniteDifferenceSpace(space)
+
+# First, dispatch on device:
+column_integral_indefinite!(
+    f::Function,
+    ᶠ∫field::Fields.Field,
+    ϕ₀ = zero(eltype(ᶠ∫field)),
+    average = (ϕ⁻, ϕ⁺) -> (ϕ⁻ + ϕ⁺) / 2,
+) = column_integral_indefinite!(
+    f,
+    ClimaComms.device(ᶠ∫field),
+    ᶠ∫field,
+    ϕ₀,
+    average,
+)
+
+#####
+##### CPU
+#####
+
+column_integral_indefinite!(
+    f::Function,
+    ::ClimaComms.AbstractCPUDevice,
+    ᶠ∫field::Fields.ColumnField,
+    args...,
+) = column_integral_indefinite_cpu!(f, ᶠ∫field, args...)
+
+column_integral_indefinite!(
+    f::Function,
+    ::ClimaComms.AbstractCPUDevice,
+    ᶠ∫field::Fields.FaceExtrudedFiniteDifferenceField,
+    args...,
+) =
+    Fields.bycolumn(axes(ᶠ∫field)) do colidx
+        column_integral_indefinite_cpu!(f, ᶠ∫field[colidx], args...)
+    end
+
+#=
+Function-based signature, solve for ϕ:
+```
+∂ϕ/∂z = f(ϕ,z)
+(ᶠϕ^{k+1}-ᶠϕ^{k})/ᶜΔz = ᶜf(ᶜϕ̄,ᶜz)
+ᶜϕ̄ = (ϕ^{k+1}+ϕ^{k})/2
```

Comment:
Why is this called `ᶜϕ̄`? (with the minus?)

Reply:
It should just be an overbar? (I made it with `\^c<tab>\phi<tab>\bar<tab>`)

Reply:
Ah! shouldn't be \bar\phi then? 
![image](https://github.com/CliMA/ClimaCore.jl/assets/9167485/ff318ead-d9db-442c-a085-3d114f6dfda4)
(When I copy and paste the symbol on GitHub, the bar gets lost)

Reply:
I don't think so? Here's what the two look like for me (left is `\bar\phi`, right is `\phi\bar`):

<img width="146" alt="Screen Shot 2023-10-20 at 9 19 39 AM" src="https://github.com/CliMA/ClimaCore.jl/assets/1880641/5e5efaed-0e80-43d9-be5b-45c4fe175455">

I'm actually surprised that `\bar\phi` puts the bar on phi here. For example, if you do `a\bar\phi`, you'll see that the `bar` goes to `a`, and not `\phi`.

Reply:
In LaTeX you put the \bar before the symbol, so I was expecting it to be the same, and if you do `\bar a`, you get a with a bar on top.

On my system, `\^c<tab>\phi<tab>\bar<tab>`, looks like:
![image](https://github.com/CliMA/ClimaCore.jl/assets/9167485/09b09428-cbce-4b6a-b0ec-e2c0a25b85f1)

Reply:
Yeah, I think this works a bit differently compared to LaTeX in this regard